File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        Set<Integer> setrow=new HashSet<>();
        Set<Integer> setcol=new HashSet<>();
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[i].length;j++){
                if(matrix[i][j]==0){
                    setrow.add(i);
                    setcol.add(j);
                }
            }
        }
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[i].length;j++){
                if(setrow.contains(i) || setcol.contains(j))
                matrix[i][j]=0;
            }
        }
    }
}

File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        setrow=set()
        setcol=set()
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if matrix[i][j]==0:
                    setrow.add(i)
                    setcol.add(j)
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if i in setrow or j in setcol:
                    matrix[i][j]=0